contracts/oracles.cairo

```diff
@@ -0,0 +1,96 @@
+%lang starknet
+
+from starkware.cairo.common.pow import pow
+from starkware.cairo.common.math_cmp import is_le
+from starkware.cairo.common.math import unsigned_div_rem
+from starkware.cairo.common.bool import TRUE, FALSE
+
+from contracts.Math64x61 import (
+    Math64x61_fromFelt,
+    Math64x61_div,
+    Math64x61_INT_PART,
+    Math64x61_add,
+    Math64x61_sqrt,
+)
+
+from contracts._cfg import EMPIRIC_ORACLE_ADDRESS, EMPIRIC_AGGREGATION_MODE
+
+# List of available tickers:
+#  https://docs.empiric.network/using-empiric/supported-assets
+
+# Contract interface copied from docs
+@contract_interface
+namespace IEmpiricOracle:
+    func get_value(key : felt, aggregation_mode : felt) -> (
+        value : felt, decimals : felt, last_updated_timestamp : felt, num_sources_aggregated : felt
+    ):
+    end
+end
+
+# Function for iterative division
+func iter_div_64x61{range_check_ptr}(x : felt, y : felt) -> (res : felt):
```

Comment:
put this in a separate module... module could be called something like `math_64x61_extended`... call this function `Math64x61_div_imprecise`